x-pack/plugins/apm/public/components/shared/backend_link.tsx

```diff
@@ -13,8 +13,19 @@ import { useApmRouter } from '../../hooks/use_apm_router';
 import { truncate } from '../../utils/style';
 import { ApmRoutes } from '../routing/apm_route_config';
 import { SpanIcon } from './span_icon';
+import { TruncateWithoutTooltip } from './truncate_with_tooltip';
 
-const StyledLink = euiStyled(EuiLink)`${truncate('100%')};`;
+const StyledLink = euiStyled(EuiLink)`min-width: 0`;
+
+const truncateAnchorClassname = '_apm_truncate_anchor_';
+
+const TruncationWrapper = euiStyled.div`
+  width: 50px;
```

Comment:
How do you determine the width?

Reply:
Here, some fixed width is required for ellipsis truncation to work. 
Reference : https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/

Reply:
The `<BackendLink />` is a shared component, there might be an issue by setting the `width:50px`.
Can you please make sure how it affects the components that use `<BackendLink />`

Reply:
Hey @kpatticha I have already checked this, I do not find this affects other components. Only service and dependency sections use this, and I have handled the scenarios accordingly.

Reply:
One quick and easy way to find where the component is used, you can run  the following command from APM folder (kibana/x-pack/plugins/apm)
`grep -rn "<BackendLink" .`
```
./public/components/app/backend_inventory/backend_inventory_dependencies_table/index.tsx:66:        <BackendLink
./public/components/app/service_overview/service_overview_dependencies_table/index.tsx:85:          <BackendLink
./public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/span_flyout/sticky_span_properties.tsx:108:            <BackendLink
./public/components/shared/backend_link.stories.tsx:30:  return <BackendLink {...args} />;

```

The given width works for the dependency section in the service overview page (`services/SERVICE_NAME/overview`) but when I check the dependencies table while I have enough space,  it truncates the dependency name
![Screenshot 2022-03-31 at 14 47 46](https://user-images.githubusercontent.com/3369346/161058524-4d5c950b-9291-4965-bb4f-05fd468b6659.png)

The same applies for the `span_flyout`.

https://user-images.githubusercontent.com/3369346/161059244-331d029e-fbd8-4898-8eae-3496ed29c6c3.mov